### PR TITLE
Add support for getting end of region from END= tag in VCF INFO field and file offset tests

### LINKED ISSUE #4: Variants with symbolic ALT and length defined by INFO END not getting queried
This happens when the ALT is symbolic and the length of the variant is defined by END in the INFO field. For example:
```
#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO
22	16050654	.	A	<CN3>	.	PASS	SVTYPE=CNV;END=16063474
```
The failing test is:
```js
it('can fetch a CNV with length defined by END in INFO field', async () => {
  const f = new TabixIndexedFile({
    path: require.resolve('./data/CNVtest.vcf.gz'),
  })

  const lines = new RecordCollector()
  await f.getLines('22', 16063470, 16063480, lines.callback)
  expect(lines.length).toEqual(1)
})
```
I checked and the `tabix` command line utility does retrieve this variant with this same VCF and index:
```
$ tabix CNVtest.vcf.gz 22:16063470-16063480 | wc -l
1
```

EDIT: you can find this failing test and the VCF/TBI in the `cnv_query` branch.

### PR DESCRIPTION
**Pull request by garrettjstevens**
This only works if the tabix index has the format as "VCF" in the metadata. One of the VCFs in our test data used the "generic" format instead and didn't work, so I regenerated all our index files just to make sure they are correct.

The best way to be sure the the index has the right format is to use tabix -p vcf my.vcf.gz, although if your file ends in .vcf.gz it should auto-detect it correctly anyway.

Resolves #4

Also adds tests (and makes one fix so they pass) to make sure the fileOffset used in the line callbacks is correct.

### CODE REVIEW

File: src/tabixIndexedFile.js [hunk 1]
```diff
@@ -291,9 +290,18 @@ class TabixIndexedFile {
             // if we have no end, we assume the feature is 1 bp long
             if (startCoordinate + 1 <= regionStart) return { overlaps: false }
           }
+        } else if ((format === 'VCF') & (currentColumnNumber === 4)) {
```
Comment (rbuels) [inline src/tabixIndexedFile.js]: I think you want `&&` there

File: src/tabixIndexedFile.js [hunk 2]
```diff
@@ -268,16 +264,19 @@ class TabixIndexedFile {
...
+    if (format === 'VCF') end = 8
+    const maxColumn = Math.max(ref, start, end)
 
     // this code is kind of complex, but it is fairly fast.
     // basically, we want to avoid doing a split, because if the lines are really long
     // that could lead to us allocating a bunch of extra memory, which is slow
 
     let currentColumnNumber = 1 // cols are numbered starting at 1 in the index metadata
     let currentColumnStart = 0
+    let refSeq = ' '
```
Comment (rbuels) [inline src/tabixIndexedFile.js]: why a single space?

**Comment by rbuels** [comment]
it looks like this incorporates #5 as well, do you want to just close #5 and merge its verbiage into here?

File: src/tabixIndexedFile.js (hunk 1 above)
Comment (garrettjstevens) [reply src/tabixIndexedFile.js]: Yup. Whoops.

File: src/tabixIndexedFile.js (hunk 2 above)
Comment (garrettjstevens) [reply src/tabixIndexedFile.js]: Holdover from a different way I tried of doing this. Just removed it.

**Comment by garrettjstevens** [comment]
Added one more small change. Returns the startCoordinate from the lineCheck when possible so that the downstream can know more quickly that it can stop processing when it's passed the query end.

**Pull request merged**